src/logic/safe/store/reducer/safe.ts

```diff
@@ -51,7 +51,8 @@ export default handleActions(
       return state.updateIn(
         ['safes', safeAddress],
         makeSafe({ name: 'LOADED SAFE', address: safeAddress }),
-        (prevSafe) => prevSafe.merge(safe),
+        // `mergeDeep` because with `merge` nested data gets overwritten
+        (prevSafe) => prevSafe.mergeDeep(safe),
```

Comment:
Adding this mergeDeep activates the owner list multiply bug. You can check it  going to settings/owners and try to full reload the app.

Also visible in transaction list when expanding details

Reply:
ouch! :man_facepalming:

Reply:
Dani, I may have a working solution but ended up solving it using a `sleep`, just for testing purposes. I guess that with a piece of context information (a flag) I will have it properly working.

Will upload changes tomorrow.

Reply:
@dasanra, well I ended up with the not so elegant solution.

I'll later create a draft PR with the current status my work on fixing the loading process. I need to make a final review of my notes.

BTW, I think it's about time to start thinking about adding XState, as Nico suggested. This is a really good scenario to start with it. Thoughts?